### DISCUSSION TITLE
Add initContainers option in helm chart

### DISCUSSION THREAD

**Discussion by aturpin1789** [jertel/elastalert2#1134] [Ideas]
Hello

I would like to propose adding the possibilities to add initContainers configuration to helm chart.
My goal is to create a container that will pull my rules directory from my gitlab repo and then apply the transformation configured in helm release config file.
I could add another container to do the job but, firstly I can not be sure that rules will be loaded before the transformation are applied and secondly, I do not want a one-shot container created.

Is this an interesting ideas ?

**Comment by jertel** [comment]
I am having trouble visualizing a "transformation configured in helm release config file". Can you elaborate?

ElastAlert 2 auto-detects rule changes, so even if your transformation was slow to apply the changes will still be quickly updated in the running ElastAlert 2 process.

You can use the `rulesVolume` chart parameter, along with the other `extra*` chart parameters, to share a rules directory volume between your custom container and the ElastAlert 2 container.

**Reply by aturpin1789** [reply]
I would like to use the `extraConfigOptions` in Helm chart config to add global config to my rules, nevertheless it will be apply only on rules in the volume I specified in `rulesVolume` and I can't load the rules from my gitlab in this volume before those extra config options will be added but I could with `initContainers` as it is action before all the other actions

**Reply by jertel** [reply]
I'd probably need to see a specific example of how a global config change is applied to all rules in `extraConfigOptions` to fully understand what you're doing. But it's not important. If you want to add an `initContainers` section to the helm chart, you can submit a PR. Make sure you follow the project's contribution guidelines linked in #11 and specifically for the helm chart changes use the same syntax and documentation patterns that are already in place.
